framework/core/hpp_device.cpp

```diff
@@ -0,0 +1,29 @@
+/* Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <core/hpp_device.h>
+
+namespace vkb
+{
+namespace core
+{
+vk::Device HPPDevice::get_handle() const
```

Comment:
vkb::core::device is not a virtual class, need to add virtual to be able to override it.

Reply:
Similar to HPPBuffer/vkb::core::Buffer, HPPDevice (now) derives protectively from vkb::Device. So no need to make vkb::Device virtual, as HPPDevice does not override anything from it.

Reply:
I've changed HPPDevice to its bar minimum now, making it a wrapper class that simply hides all the vkb::Device stuff.